### PR TITLE
Improved perf of SingleThreadEventExecutor

### PR DESCRIPTION
**Pull request by nayato**
Motivation:
Impove performance of execution model

Modifications:
- switched SemaphoreSlim for ManualResetEventSlim to track empty vs non-empty states for executor's queue

extras
- removing wait for event loop shutdown (reduces test run time)
- adding benchmarking
- extracted common test utilities into Tests.Common project
- fixing ReSharper settings for ReSharper 10

Results:
4x improvement on in-loop scheduling, 2x improvement on out-of-liio scheduling:
```
STEE in loop (new)
        624ms   1,406,159Kc (G0= 114, G1=   0, G2=   0)
STEE in loop (old)
      2,719ms   6,205,423Kc (G0= 114, G1=   0, G2=   0)
STEE out of loop (new)
      1,464ms   6,559,838Kc (G0=  95, G1=   0, G2=   0)
STEE out of loop (old)
      3,869ms  12,331,986Kc (G0=  38, G1=  28, G2=   3)
```

### CODE REVIEW

File: test/DotNetty.Tests.Common/TestBase.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+{
+    using System;
+    using System.Diagnostics.Tracing;
+    using DotNetty.Common.Internal.Logging;
+    using Microsoft.Practices.EnterpriseLibrary.SemanticLogging;
+    using Xunit.Abstractions;
+
+    public abstract class TestBase : IDisposable
+    {
+        protected readonly ITestOutputHelper Output;
```
Comment (Aaronontheweb) [inline test/DotNetty.Tests.Common/TestBase.cs]: Just for my benefit here, what's the purpose of this new base class and the `ITestOutputHelper` ?

Comment (nayato) [inline test/DotNetty.Tests.Common/TestBase.cs]: that's the only way I know of to do test output with XUnit.

Comment (Aaronontheweb) [inline test/DotNetty.Tests.Common/TestBase.cs]: Test output for what? You mean just writing additional data to the XUnit output file?

Comment (nayato) [inline test/DotNetty.Tests.Common/TestBase.cs]: pretty much. When ran from within VS, it shows up in test output.

Comment (Aaronontheweb) [inline test/DotNetty.Tests.Common/TestBase.cs]: Got it - makes sense

**Pull request merged**